src/components/ToDo.tsx

```diff
@@ -0,0 +1,179 @@
+
+  function handleClick() {
+    setIsLoading(true);
+    updateTodo({
+      id: todo.id,
+      completed: !todo.completed,
+      title: todo.title,
+      userId: todo.userId,
+    })
+      .then(() => {
+        refreshLIst();
+      })
+      .catch(() => dispatch({ type: ACTIONS.SET_ERROR, payload: 'Unable to toggle a todo' }))
+  }
+
+  function handleChanges() {
+    console.log(editingtoDoData, 'data');
+
+    if (editingtoDoData === '') {
+      deleteItem(todo.id);
+    }
+    else {
+      setIsLoading(true);
+      updateTodo({
+        id: todo.id,
+        completed: todo.completed,
+        userId: todo.userId,
+        title: editingtoDoData,
+      })
+        .then(() => refreshLIst())
+        .catch(() => dispatch({ type: ACTIONS.SET_ERROR, payload: 'Unable to update a todo' }))
+      setIsEditing(false);
+    }
+  }
+
+  function handleEnter(e: React.KeyboardEvent<HTMLInputElement>) {
+    if (e.nativeEvent.code === 'Enter' && e.target.value === '') {
+      deleteItem(todo.id);
+      setIsEditing(false);
+    } else if (e.nativeEvent.code === 'Enter') {
+      handleChanges();
+    }
+
+  }
+  return (
+    <div className={classNames('todo', {
+      'completed': todo.completed,
+      editing: isEditing,
+    })}>
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={handleClick}
+        />
+      </label>
+
+      {isLoading && (
+
+        <div className="modal overlay is-active">
+          <div className="modal-background has-background-white-ter" />
+          <div className="loader" />
+        </div>
+      )}
+      { (state.isLoading && todo.completed) && (
+        <div className="modal overlay is-active">
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
+      )}
+      { (state.toggleAll === 'completed' && todo.completed) && (
+        <div className="modal overlay is-active">
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
+      )}
+      { (state.toggleAll === 'active' && !todo.completed) && (
+        <div className="modal overlay is-active">
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
+      )}
+      {isEditing ? (
+        <form>
+          <input
+            ref={editingTodo}
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value={editingtoDoData}
+            onChange={(e) => setEditingToDoData(e.target.value)}
+            onBlur={handleChanges}
+            onKeyDown={handleEnter}
+          />
+        </form>
+      ) : (
+        <span
+          className="todo__title"
+          onDoubleClick={(e) => handleDoubleClickEdit(e)}
```

Comment:
Will work same, but less callbacks
```suggestion
          onDoubleClick={handleDoubleClickEdit}
```